rerun_cpp/src/rerun/component_batch.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <array>
+#include <vector>
+
+#include "data_cell.hpp"
+#include "result.hpp"
+
+namespace rerun {
+    /// Generic list of components that are contiguous in memory.
+    ///
+    /// Does *not* own the data, user is responsible for the lifetime independent of how it was
+    /// passed in.
+    template <typename ComponentType>
+    class ComponentBatch {
+      public:
+        const ComponentType* data;
+        size_t size;
```

Comment:
Here and everywhere else: `size` is very ambiguous as well as inconsistent with the rest of the codebase -- we try to stick to `num_instances` everywhere.

```suggestion
        size_t num_instances;